plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -0,0 +1,546 @@
+        jenkinsWarFile = new File(cfg.getJenkinsWar());
+        installedPluginVersions = new HashMap<>();
+        bundledPluginVersions = new HashMap<>();
+        refDir = cfg.getPluginDir();
+        this.cfg = cfg;
+        allSecurityWarnings = new ArrayList<>();
+    }
+
+
+    public void start() {
+        if (!refDir.mkdir()) {
+            System.out.println("Unable to create plugin directory");
+        }
+
+        jenkinsVersion = getJenkinsVersion();
+        checkVersionSpecificUpdateCenter();
+
+        String url;
+
+        getSecurityWarnings();
+
+        if (cfg.hasShowAllWarnings()) {
+            for (int i = 0; i < allSecurityWarnings.size(); i++) {
+                SecurityWarning securityWarning = allSecurityWarnings.get(i);
+                System.out.println(securityWarning.getName() + " - " + securityWarning.getMessage());
+            }
+        }
+
+        createLocks(plugins);
+
+        List<String> bundledPlugins = bundledPlugins();
+        List<String> installedPlugins = installedPlugins();
+
+        downloadPlugins(plugins);
+
+        writeFailedPluginsToFile();
+
+        //clean up locks
+
+    }
+
+
+    public void getSecurityWarnings() {
+        JSONObject updateCenterJson = getUpdateCenterJson();
+        JSONArray warnings = updateCenterJson.getJSONArray("warnings");
+
+        for (int i = 0; i < warnings.length(); i++) {
+            JSONObject warning = warnings.getJSONObject(i);
+            String warningType = warning.getString("type");
+            if (!warningType.equals("plugin")) {
+                continue;
+            }
+            String warningId = warning.getString("id");
+            String warningMessage = warning.getString("message");
+            String warningName = warning.getString("name");
+            String warningUrl = warning.getString("url");
+
+            SecurityWarning securityWarning = new SecurityWarning(warningId, warningMessage, warningName, warningUrl);
+            JSONArray warningVersions = warning.getJSONArray("versions");
+            for (int j = 0; j < warningVersions.length(); j++) {
+                JSONObject warningVersion = warningVersions.getJSONObject(j);
+                String lastVersion = "";
+                if (warningVersion.has("lastVersion")) {
+                    lastVersion = warningVersion.getString("lastVersion");
+                }
+                String pattern = warningVersion.getString("pattern");
+                securityWarning.addSecurityVersion(lastVersion, pattern);
+            }
+            allSecurityWarnings.add(securityWarning);
+        }
+    }
+
+
+    public void checkVersionSpecificUpdateCenter() {
+        //check if version specific update center
+        if (!StringUtils.isEmpty(jenkinsVersion)) {
+            JENKINS_UC_LATEST = new StringBuilder(JENKINS_UC).append(jenkinsVersion).toString();
+
+            try (CloseableHttpClient httpclient = HttpClients.createDefault();) {
```

Comment:
Should only need semi colons if closing multiple resources
```suggestion
            try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
```